http/src/main/scala/org/broadinstitute/dsde/workbench/leonardo/util/GKEInterpreter.scala

```diff
@@ -187,138 +124,330 @@ class GKEInterpreter[F[_]: Parallel: ContextShift](
         )
         .compile
         .lastOrError
-      _ <- if (lastOp.isDone) F.unit
+      _ <- if (lastOp.isDone)
+        logger.info(
+          s"Delete nodepool operation has finished for nodepool ${params.nodepoolId} | trace id: ${ctx.traceId}"
+        )
       else
-        logger.error(s"Delete nodepool operation has failed or timed out for nodepool ${result.nodepoolId}") >> F
-          .raiseError[Unit](NodepoolDeletionException(result.nodepoolId))
-      _ <- nodepoolQuery.markAsDeleted(result.nodepoolId, ctx.now).transaction
+        logger.error(
+          s"Delete nodepool operation has failed or timed out for nodepool ${params.nodepoolId} | trace id: ${ctx.traceId}"
+        ) >>
+          F.raiseError[Unit](NodepoolDeletionException(params.nodepoolId))
+      _ <- nodepoolQuery.markAsDeleted(params.nodepoolId, ctx.now).transaction
     } yield ()
 
-  def deleteAndPollApp(getAppResult: GetAppResult)(
+  override def deleteAndPollApp(params: DeleteAppParams)(
     implicit ev: ApplicativeAsk[F, AppContext]
   ): F[Unit] =
-    //TODO: this should actually monitor the deletion went through successfully before updating the status
     for {
-      now <- nowInstant
-      _ <- kubeService.deleteNamespace(getAppResult.cluster.getGkeClusterId,
-                                       KubernetesNamespace(getAppResult.app.appResources.namespace.name))
-      _ <- appQuery.markAsDeleted(getAppResult.app.id, now).transaction
+      ctx <- ev.ask
+      _ <- logger.info(
+        s"Beginning app deletion for app ${params.appId} in project ${params.googleProject} | trace id: ${ctx.traceId}"
+      )
+      dbAppOpt <- KubernetesServiceDbQueries.getActiveFullAppByName(params.googleProject, params.appName).transaction
+      dbApp <- F.fromOption(dbAppOpt, AppNotFoundException(params.googleProject, params.appName, ctx.traceId))
+      // TODO helm delete app and monitor for deletion
+      _ <- kubeService.deleteNamespace(dbApp.cluster.getGkeClusterId,
+                                       KubernetesNamespace(dbApp.app.appResources.namespace.name))
+      _ <- logger.info(
+        s"Delete app operation has finished for nodepool ${params.appId} | trace id: ${ctx.traceId}"
+      )
+      _ <- appQuery.markAsDeleted(dbApp.app.id, ctx.now).transaction
     } yield ()
+
+  private[util] def getSecrets(namespace: NamespaceName): F[List[KubernetesSecret]] =
+    config.ingressConfig.secrets.traverse { secret =>
+      for {
+        secretFiles <- secret.secretFiles
+          .traverse { secretFile =>
+            for {
+              bytes <- readFileToBytes(secretFile.path, blocker)
+            } yield (secretFile.name, bytes.toArray)
+          }
+          .map(_.toMap)
+      } yield KubernetesSecret(
+        namespace,
+        secret.name,
+        secretFiles,
+        KubernetesSecretType.Generic
+      )
+    }
+
+  private[util] def getGoogleNodepool(nodepool: Nodepool): com.google.container.v1.NodePool = {
```

Comment:
FWIW, the method name threw me off wondering how a Google nodepool would be different and also thought it would get nodepools from somewhere. At the same time I couldn't find a good name reflecting what the method achieves :(

Reply:
This is really building the google nodepool to pass in the create cluster request… hm maybe `buildGoogleNodepool`?

Reply:
Yea, I like `buildGoogleNodepool` better 😊